Engines/Wine/QuickScript/Quick Script/script.js

```diff
@@ -120,8 +120,12 @@ module.default = class QuickScript {
      * @param {string} environment variables
      * @returns {QuickScript} QuickScript object
      */
-    environment(environment) {
-        this._environment = environment;
+    environment(environmentFunc) {
+        if (environmentFunc && environmentFunc instanceof Function) {
+            this._environmentFunc= environmentFunc;
+        } else {
+            throw new Error(tr("The argument of environment() should be a function !"));
```

Comment:
Why not allow to pass the environment as a string also?

Reply:
Because strings aren't directory paths.

Reply:
Then I don't understand how it worked before.

Reply:
Perhaps it never worked. Because I remember earlier today I did see nvapi workaround being enabled when using old `.environment` implementation.

Reply:
It worked before because the envirnment was assumed to be a stri ng not dependent of any external variable. With this, one can construct the environment with part depedending of wine or user input.

We could also add an option to just pass a plain string instead of a function in the case the string does not depends but it can be done with:
```js
.environment((wine) => {
    return '{"DXVK_HUD" : "1" }';
})
```
I think I prefer to use a uniform solution for this.

Reply:
> I think I prefer to use a uniform solution for this.

I understand this, but I'm not sure I agree in this case.
Adding a second input type is as simple as adding another `else if` to the `.environment(...)` implementation. The benefit of this is that you:
- don't get any unused variable messages by Codacy
- make the code more concise

Reply:
Well, it is a question of choice. It does not bother me the way I propose it and it is not really that big (in comparaison of .postInstall or .preInstall generally). Always using a lambda function makes the code easier to write I guess (only one way to use and to code).

But if @Zemogiter is motivated ^^.